src/prefect/agent.py

```diff
@@ -36,7 +37,8 @@ def __init__(
                 "Provide only one of 'default_infrastructure' and 'default_infrastructure_document_id'."
             )
 
-        self.work_queues: Set[str] = set(work_queues)
+        self.work_queues: Set[str] = set(work_queues) if work_queues else set()
+        self.work_queue_regex = work_queue_regex
```

Comment:
If you update the client to accept a compiled regex as well we can compile it here which will be helpful for performance since we're using it repeatedly.

Reply:
yeah that's a good idea